pallets/nft-manager/src/lib.rs

```diff
@@ -260,19 +276,20 @@ pub mod pallet {
     #[pallet::storage]
     #[pallet::getter(fn nft_batches)]
     pub type NftBatches<T: Config> =
-        StorageMap<_, Blake2_128Concat, NftBatchId, Vec<NftId>, ValueQuery>;
+        StorageMap<_, Blake2_128Concat, NftBatchId, BoundedVec<NftId, T::BatchBound>, ValueQuery>;
 
     /// A mapping between the external batch id and its corresponding NtfInfoId
     #[pallet::storage]
     #[pallet::getter(fn batch_info_id)]
     pub type BatchInfoId<T: Config> =
         StorageMap<_, Blake2_128Concat, NftBatchId, NftInfoId, ValueQuery>;
 
+    /// TODO: convert the key to BoundedVec once a runtime with the storage migration is applied.
     /// A mapping between an ExternalRef and a flag to show that an NFT has used it
     #[pallet::storage]
     #[pallet::getter(fn is_external_ref_used)]
     pub type UsedExternalReferences<T: Config> =
-        StorageMap<_, Blake2_128Concat, Vec<u8>, bool, ValueQuery>;
+        StorageMap<_, Blake2_128Concat, WeakBoundedVec<u8, NftExternalRefBound>, bool, ValueQuery>;
```

Comment:
Unless I am missing something, it looks like we have 2 slightly different types for external refs: 
 - When we store them as part of the nft they use BoundedVec
 - When we store then in the `UsedExternalReferences` storage they use WeakBoundedVec
 Is this intentional?

Reply:
yes, the storage representation is the same and we use it to migrate data. the next release will rectify things and make everything a "BoundedVector"